### DISCUSSION TITLE
Monorepo

### DISCUSSION THREAD

**Discussion by faddat** [osmosis-labs/osmosis#653] [Ideas]
Osmosis is a product.  But the product is all over in many repositories.  Here's how many:

github.com/osmosis-labs/osmosis
github.com/osmosis-labs/frontend
github.com/osmosis-labs/assetlists

I think there's more too. 

But what if the product were a single repository?

**Comment by daniel-farina** [comment]
Mono repos are great sometimes, however I'm not a big fan of Mono repos in this particular case. 
I think Osmosis is more than just the Dex, the dex front-end is just the best example of what we can accomplish.  This is definitely an item to keep on the backlog for discussion but I would mark it as low priority for now. 

I'll convert this issue to a discussion for now. I would like to give discussions a chance.  Great way to discuss ideas while keeping issues lean. 

@faddat What do you think would be the main benefit of doing a mono-repo? I would love to learn more and hear your opinion.

**Reply by faddat** [reply]
Hi Daniel, I'm sorry that I didn't see this till now, 5 months later!

I just wrote another issue describing the benefits of a monorepo for osmosis:

https://github.com/osmosis-labs/osmosis/issues/1269

For me, it would make it much easier to contribute to more of osmosis, and make the codebase and versioning easier to understand.  The gitlab part of that issue is purely optional.
